### PR TITLE
Add importable images

### LINKED ISSUE #101: Add some images that can be loaded via the public API
We now have one image in the repo: ``demo/data/aero.png``. We can't use that in docstring examples however, because it's not importable. Nicer docstring examples instead of just the return values of some random numbers would be very good to have. 
We could easily add ``scipy.misc.face`` and ``scipy.misc.ascent`` next to ``aero``.

### PR DESCRIPTION
**Pull request by aaren**
Fixes #101.

Usage:

```python
import pywt.data

aero = pywt.data.aero()
ascent = pywt.data.ascent()
face = pywt.data.face()
```

I just pulled code and data from scipy for ascent and face. I can add in more images if wanted.

@grlee77 this is pretty simple so can we get it in for 0.4?

### CODE REVIEW

**Comment by aaren** [comment]
I've also added the ECG time-series that was in `pywt/demo/data`.

**Comment by grlee77** [comment]
I would be fine with including this for 0.4.0.  

Can you also update the demos that were using `ecg` or `aero` to import from `pywt.misc` so that the `demos/data` folder can be removed?

I guess you chose `pywt.misc` to match scipy's convention?  I think the name `pywt.data` would be clearer in intent, but let's see what others think.  `scikit-image` stores their images in `skimage.data` so there is some precedent for that naming scheme as well.

**Comment by aaren** [comment]
Yes, will update demos.

`pywt.misc` was arbitrary. I nearly changed it to `pywt.data`. I'll let someone else cast the vote on that.

**Comment by aaren** [comment]
I've just changed it to `pywt.data` as that's a much more sensible name than the vague `misc`.

Have removed the old data files and updated the demo scripts.

**Comment by rgommers** [comment]
nice. and +1 for ``pywt.data``

File: demo/swt2.py [hunk 1]
```diff
@@ -6,11 +6,10 @@
 from PIL import Image
```
Comment (rgommers) [inline demo/swt2.py]: This PIL import can be removed.

**Comment by rgommers** [comment]
The binary file approach looks good, and removes a dependency on PIL which is helpful. My only question on that is why not use the ``.npy`` extension instead of ``.dat``?

**Comment by rgommers** [comment]
Also, if we need image import somewhere in docs somewhere, I'd be in favor of depending on ``imageio`` instead of ``PIL``. ``PIL`` is awful for scientific image formats, and riddled with bugs.

**Comment by aaren** [comment]
Data formats:

- ascent: pickled list
- face: bz2 compressed binary data
- aero: numpy npy
- ecg: numpy npy

`ascent` and `face` were copied directly from scipy. Not sure why they use the formats that they do. I chose `.dat` for consistency. Now I'm thinking reformat everything to `.npy` and use that as the extension.

Do we actually want all three channels of `face`? We'd save ourselves 1MB of repo size if it was just saved as gray. The data directory is currently 2.3MB.

**Comment by aaren** [comment]
Comparative sizes of different formats (using `np.savez_compressed` for `.npz`):

```
516K	ascent.dat
2.1M	ascent.npy
256K	ascent.npz
```

```
1.6M	face.dat
2.3M	face.npy
1.9M	face.npz
```

```
260K	aero.dat
260K	aero.npy
224K	aero.npz
```

```
260K	camera.npy
160K	camera.npz
```

On this basis I think we should 1) replace `face` with `camera` (from scikit-image), as the latter is quite a bit smaller and 2) use compressed `.npz` as the storage format.

Please let me know if I'm overthinking this! It seems a shame to inflate the repo size unnecessarily.

**Comment by grlee77** [comment]
+1 on using either .npy or .npz for all of these
+1 on using camera instead of face

**Comment by grlee77** [comment]
`scipy.misc.aero` is only a bigger file than the others because it is stored as `np.int64`.  However, the image values are all in [0-255] so it could be stored as `np.uint8` instead.  

aero, camera and ascent are then all 512x512 grayscale integer data, but since they are all pretty small I don't have a problem with keeping all three to provide some variety.

**Comment by aaren** [comment]
Have reformatted to npz / npy and replaced face with camera. 

To avoid a messy rebase I edited the commits where I first added the image data files and replaced them with empty stubs, then added the actual data at the end. I think this should avoid any binary data being stored in the history.

**Comment by grlee77** [comment]
great.  LGTM.  I just added some brief documentation in #138 if you want to take a look.

**Pull request merged**